HyperVExtension/src/HyperVExtension/CommunicationWithGuest/WmiUtility.cs

```diff
@@ -124,4 +138,47 @@ public static bool JobCompleted(ManagementBaseObject outParams, ManagementScope
 
         return jobCompleted;
     }
+
+    public static FeatureAvailabilityKind GetHyperVFeatureAvailability()
+    {
+        try
+        {
+            var searcher = new ManagementObjectSearcher($"SELECT * FROM Win32_OptionalFeature WHERE Name = 'Microsoft-Hyper-V'");
+            var collection = searcher.Get();
+
+            foreach (var instance in collection)
+            {
+                if (instance?.GetPropertyValue("InstallState") is uint enablementState)
+                {
+                    var featureAvailability = GetAvailabilityKindFromState(enablementState);
+
+                    _log.Information($"Found Hyper-V feature with enablement state: '{featureAvailability}'");
+                    return featureAvailability;
+                }
+            }
+        }
+        catch (Exception ex)
+        {
+            // We'll handle cases where there are exceptions as if the feature does not exist.
+            _log.Error(ex, $"Error attempting to get the Hyper-V feature state");
+        }
+
+        _log.Information($"Unable to find Hyper-V feature");
+        return FeatureAvailabilityKind.Unknown;
+    }
+
+    private static FeatureAvailabilityKind GetAvailabilityKindFromState(uint state)
+    {
+        switch (state)

```

Comment:
I think if you assign values to the enum, C# can do a conversion from uint to enum.

Reply:
Output
```
Enabled
Disabled
Absent
Unknown
Enabled
Disabled
Absent
Unknown
```

You would need to check for the error case.

Reply:
Technically, not implicit, but can get rid of the method.

Reply:
+1 for this one. Shouldn't need to use this method.

Reply:
Hmm what's the benefit to using Enum.Parse in this case over the switch statement exactly? They're both still ultimately method calls. Enum.Parse is using reflection under the hood so not sure if its faster than simply checking the switch statement.

Reply:
I agree with Branden: no need to use less performant solution just because it's it a couple lines shorter. And since you'd need to check for "> Unknown" you'd wrap it in a method anyway. If it seems too long with "switch" here is a shorter variant:
```
if (state > (uint)FeatureAvailabilityKind.Unknown)
{
    return FeatureAvailabilityKind.Unknown;
}

return (FeatureAvailabilityKind)state;
```